geotrek/common/forms.py

```diff
@@ -79,6 +77,32 @@ def __init__(self, *args, **kwargs):
             self.deep_remove(self.helper.fieldslayout, 'published')
         if 'review' in self.fields and self.instance and self.instance.any_published:
             self.deep_remove(self.helper.fieldslayout, 'review')
+        if 'structure' in self.fields:
+            if self.user.has_perm('authent.can_bypass_structure'):
+                if not self.instance.pk:
+                    self.fields['structure'].initial = self.user.profile.structure
+            else:
+                for name, field in self.fields.items():
+                    self.filter_related_field(name, field)
+                del self.fields['structure']
+
+    def clean(self):
+        structure = self.cleaned_data.get('structure')
+        if not structure:
+            return self.cleaned_data
+
+        for name, field in self.cleaned_data.items():
+            if isinstance(field, QuerySet):
+                for value in field:
+                    self.check_structure(value, structure, name)
+            else:
+                self.check_structure(field, structure, name)
+        return self.cleaned_data
+
+    def check_structure(self, obj, structure, name):
+        if hasattr(obj, 'structure'):
+            if structure != obj.structure:
+                self.add_error(name, _("The structure given to the global object doesn't correspond"))
```

Comment:
"Please select a choice related to all structures (without brackets) or related to the structure {} (in brackets)".format(obj.structure.name) would be more explicit for end users.